internal/storage/blob/store.go

```diff
@@ -404,7 +399,16 @@ func (e *indexBuildError) Error() string {
 // s.cacheDir according to the given map 'all' and tries to build a temporary index to see if there are any errors
 // with the incoming policies/schemas. If there are no errors returns the index built and the path to the new work directory.
 func (s *Store) buildIndex(ctx context.Context, all map[string][]string) (idx index.Index, ts string, err error) {
-	ts = strconv.FormatInt(time.Now().UnixMicro(), 10)
+	tsMicro := time.Now().UnixMicro()
```

Comment:
I think that microsecond granularity is a bit too much 😄 `UnixMillis` should be more than enough.

Reply:
Does that test compare the timestamp used for this metric? I assume the problem is that the store directory is based on Unix timestamp? We should maybe add a random value as a suffix to the directory name to avoid those types of problems.

Reply:
Yes, the problem occurs when insufficient time has passed and the same (timestamp) directory is used for the next update.

I will implement your suggestion.